ipfx/sweep.py

```diff
@@ -54,22 +53,23 @@ def detect_epochs(self):
         """
 
         if "test" not in self.epochs:
-            self.epochs["test"] = ep.get_test_epoch(self.stimulus,self.sampling_rate)
+            self.epochs["test"] = ep.get_test_epoch(self._stimulus, self.sampling_rate)
         if self.epochs["test"]:
             test_pulse = True
         else:
             test_pulse = False
 
-        epoch_detectors = {
-            "sweep": ep.get_sweep_epoch(self.response),
-            "recording": ep.get_recording_epoch(self.response),
-            "experiment": ep.get_experiment_epoch(self._i, self.sampling_rate,test_pulse),
-            "stim": ep.get_stim_epoch(self.stimulus, test_pulse),
-        }
-
-        for epoch_name, epoch_detector in epoch_detectors.items():
-            if epoch_name not in self.epochs:
-                self.epochs[epoch_name] = epoch_detector
+        if "sweep" not in self.epochs:
+            self.epochs["sweep"] = ep.get_sweep_epoch(self._i)
+        self.select_epoch("sweep")
```

Comment:
Is this line necessary since the epoch gets changed to "recording" three lines down?

Reply:
yeah you're right, was a holdover from a previous version before i simplified the sequence here, i'll remove.